### PR TITLE
GS-HW: Invert rounding when shader result will get subtracted

### LINKED ISSUE #8216: [BUG]: Regression: Test Drive Eve of Destruction broken shadows.
### Describe the Bug

Since #6931 shadows no longer work on Maximum blending.
Sten mentioned that it might be an underflow issue.

### Reproduction Steps

Run the provided gs dump, crank up blending to maximum on Vulkan/OpenGL.

### Expected Behavior

Shadows should work.

### PCSX2 Revision

v1.7.4123

### Operating System

Windows 10 (64bit)

### If Linux - Specify Distro

_No response_

### CPU

i7-4770

### GPU

RX570

### GS Settings

Default.

### Emulation Settings

Default.

### GS Window Screenshots

_No response_

### Logs & Dumps

Rename to .xz.
[Test Drive Eve of Destruction Shadows.gs.zip](https://github.com/PCSX2/pcsx2/files/10795373/Test.Drive.Eve.of.Destruction.Shadows.gs.zip)

### PR DESCRIPTION
**Pull request by TellowKrinkle**
### Description of Changes
SW: ((dst - 7) & 0xf8) = (dst - 8)
Broken: (dst - (7 & 0xf8)) = (dst - 0)
Fixed: (dst - ((7 + 7) & 0xf8)) = (dst - 8)

Fixes #8216 (and it now works on all blend modes, yay)
(Also includes a few other minor fixes for things I hit while testing)

### Rationale behind Changes
Better accuracy
Improves #5246
### Suggested Testing Steps
Test all renderers
(I've only tested Metal and Vulkan so far)

### CODE REVIEW

File: bin/resources/shaders/dx11/tfx.fx [hunk 1]
```diff
@@ -757,9 +757,16 @@ void ps_color_clamp_wrap(inout float3 C)
 
 		// In 16 bits format, only 5 bits of color are used. It impacts shadows computation of Castlevania
 		if (PS_DFMT == FMT_16 && PS_BLEND_MIX == 0)
-			C = (float3)((int3)C & (int3)0xF8);
+		{
+			int3 icolor = (int3)C;
+			if (PS_ROUND_INV != 0)
```
Comment (lightningterror) [inline bin/resources/shaders/dx11/tfx.fx:762]: Need to define PS_ROUND_INV at the top like the others.

**Review by lightningterror — COMMENTED** [review]
Also need to bump shader cache.

**Comment by lightningterror** [comment]
So far found 8 that show a change/break on all renderers.
Example: 
image
image

Dumps:
[accu blend mix_p2.zip](https://github.com/PCSX2/pcsx2/files/10832202/accu.blend.mix_p2.zip)
[accu blend mix_p1.zip](https://github.com/PCSX2/pcsx2/files/10832213/accu.blend.mix_p1.zip)

**Comment by TellowKrinkle** [comment]
Simpsons was because I forgot to clamp after adding, so high values overflowed

The rest were because dither values also needed to be inverted

**Comment by TellowKrinkle** [comment]
And no clue what the difference was for that disney one.  You'll have to point it out

**Comment by lightningterror** [comment]
DX1 Tests
Happy! Happy!! Boarders in Hokkaidou Rusutsu Resort- HW Renderer issues
Master:
image
PR:
image

Baldur's Gate 2
Master:
image
PR:
image

Justice League Heroes
Master:
image
PR:
image

Snowblind also shows an improvement but it's very minor to the shadow, Shadow Tower abyss shows a difference but that game has other issues so it's hard to compare to sw renderer now until the other stuff is resolved.

**Review by lightningterror — APPROVED** [review]

**Pull request merged**